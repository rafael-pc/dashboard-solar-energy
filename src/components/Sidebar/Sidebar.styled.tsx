import styled from "styled-components";
import { theme } from "../../styles";

export const Container = styled.aside`
  a {
    text-decoration: none;
    transition: 200ms;
  }

  ul {
    padding: 0;
    list-style-type: none;
  }
`;

export const Content = styled.div`
  background-color: ${theme.colors.bg_primary};

  @media only screen and (max-width: ${theme.screen.mobile}) {
    nav {
      width: 100vw;
      height: 50px;
    }

    .logo-mobile {
      width: 35px;
      height: 35px;
      border-radius: ${theme.border.radius.xs};
      margin-top: 7px;
      margin-left: 15px;
    }
  }

  @media only screen and (min-width: ${theme.screen.mobile}) {
    display: flex;
    width: 30vw;
    height: 100vh;
    flex-direction: column;
    align-items: center;
    padding: 15px;
    text-align: center;

    .logo-mobile {
      display: none;
    }
  }
  @media only screen and (min-width: ${theme.screen.desktop}) {
    width: 20vw;
  }
`;

export const Menu = styled.ul`
  @media only screen and (max-width: ${theme.screen.mobile}) {
    display: none;
  }

  @media only screen and (min-width: ${theme.screen.mobile}) {
    #menu-icon {
      display: none;
    }
  }
`;

export const Logo = styled.img`
  width: 150px;
  height: 150px;
  margin-top: 30px;
  margin-bottom: 80px;
  border-radius: ${theme.border.radius.m};
  background-color: ${theme.colors.neutral_100};

  @media only screen and (max-width: ${theme.screen.mobile}) {
    width: 100px;
    height: 100px;
    margin-left: 15px;
  }

  @media only screen and (max-height: ${theme.screen.mobile}) {
    width: 80px;
    height: 80px;
    margin-top: 10px;
    margin-bottom: 30px;
  }
`;

export const Page = styled.li`
  .link {
    height: 40px;
    display: flex;
    align-items: center;
    gap: 20px;
    margin-bottom: 40px;
    text-align: left;
    color: ${theme.colors.neutral_100};
  }

  .chart-pie {
    margin-left: 20px;
  }

  .chart-bar {
    margin-left: 20px;
  }

  .cog {
    margin-left: 20px;
  }

  h3 {
    width: 150px;
    font-weight: ${theme.font.bold};
    font-size: ${theme.font.sizes.xxs};
    line-height: 1.2em;
  }

  @media only screen and (max-width: ${theme.screen.large_desktop}) {
    gap: 15px;
  }

  @media only screen and (max-height: ${theme.screen.mobile}) {
    gap: 10px;

    h3 {
      font-size: ${theme.font.sizes.xxxxs};
    }

    .link {
      margin-bottom: 10px;
    }

    .chart-pie {
      margin-left: 6px;
    }
    .chart-bar {
      margin-left: 6px;
    }
    .cog {
      margin-left: 6px;
    }
  }
`;
