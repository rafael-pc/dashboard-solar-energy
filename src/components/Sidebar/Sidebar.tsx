import { FC } from "react";
import { Link } from "react-router-dom";
import { FaChartPie, FaChartBar, FaCog } from "react-icons/fa";
import * as S from "./Sidebar.styled";
import LogoMobile from "../../assets/logoMobile.png";
import LogoImage from "../../assets/logo.png";

const Sidebar: FC = () => {
  return (
    <S.Container>
      <S.Content>
        <nav>
          <Link to="/dashboard" className="link">
            <S.Logo src={LogoMobile} alt="logo" className="logo-mobile"/>
          </Link>
          <S.Menu>
            <S.Logo src={LogoImage} alt="logo" className="logo"/>
            <S.Page>
              <Link to="/dashboard" className="link">
                <FaChartPie className="chart-pie" />
                <h3>Dashboard</h3>
              </Link>
            </S.Page>
            <S.Page>
              <Link to="/units" className="link">
                <FaChartBar className="chart-bar" />
                <h3>Unidade consumidora</h3>
              </Link>
            </S.Page>
            <S.Page>
              <Link to="/register_month" className="link">
                <FaCog className="cog" />
                <h3>Cadastro de enÃ©rgia gerada</h3>
              </Link>
            </S.Page>
          </S.Menu>
        </nav>
      </S.Content>
    </S.Container>
  );
};

export default Sidebar;
