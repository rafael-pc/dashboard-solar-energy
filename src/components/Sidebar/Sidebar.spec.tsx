import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Sidebar from './Sidebar';
import "@testing-library/jest-dom";

describe('Sidebar', () => {
  it('renders correctly', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Sidebar />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();

    expect(screen.getByTestId('logo')).toBeInTheDocument();
    expect(screen.getByText('Dashboard')).toBeInTheDocument();
    expect(screen.getByText('Unidade consumidora')).toBeInTheDocument();
    expect(screen.getByText('Cadastro de en√©rgia gerada')).toBeInTheDocument();
  });
});
