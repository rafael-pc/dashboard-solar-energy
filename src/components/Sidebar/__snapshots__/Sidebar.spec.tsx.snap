// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar renders correctly 1`] = `
<DocumentFragment>
  <aside
    class="sc-bcPKhP hQtlFs"
  >
    <div
      class="sc-grXZZQ fmGhmA"
    >
      <nav>
        <a
          class="link"
          href="/dashboard"
        >
          <img
            alt="logo"
            class="sc-hLclGa kfTuZa logo-mobile"
            src="test-file-stub"
          />
        </a>
        <ul
          class="sc-dkjKgF JrqNC"
        >
          <img
            alt="logo"
            class="sc-hLclGa kfTuZa logo"
            data-testid="logo"
            src="test-file-stub"
          />
          <li
            class="sc-eDnVMP bhYUgY"
          >
            <a
              class="link"
              href="/dashboard"
            >
              <svg
                class="chart-pie"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 544 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M527.79 288H290.5l158.03 158.03c6.04 6.04 15.98 6.53 22.19.68 38.7-36.46 65.32-85.61 73.13-140.86 1.34-9.46-6.51-17.85-16.06-17.85zm-15.83-64.8C503.72 103.74 408.26 8.28 288.8.04 279.68-.59 272 7.1 272 16.24V240h223.77c9.14 0 16.82-7.68 16.19-16.8zM224 288V50.71c0-9.55-8.39-17.4-17.84-16.06C86.99 51.49-4.1 155.6.14 280.37 4.5 408.51 114.83 513.59 243.03 511.98c50.4-.63 96.97-16.87 135.26-44.03 7.9-5.6 8.42-17.23 1.57-24.08L224 288z"
                />
              </svg>
              <h3>
                Dashboard
              </h3>
            </a>
          </li>
          <li
            class="sc-eDnVMP bhYUgY"
          >
            <a
              class="link"
              href="/units"
            >
              <svg
                class="chart-bar"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M332.8 320h38.4c6.4 0 12.8-6.4 12.8-12.8V172.8c0-6.4-6.4-12.8-12.8-12.8h-38.4c-6.4 0-12.8 6.4-12.8 12.8v134.4c0 6.4 6.4 12.8 12.8 12.8zm96 0h38.4c6.4 0 12.8-6.4 12.8-12.8V76.8c0-6.4-6.4-12.8-12.8-12.8h-38.4c-6.4 0-12.8 6.4-12.8 12.8v230.4c0 6.4 6.4 12.8 12.8 12.8zm-288 0h38.4c6.4 0 12.8-6.4 12.8-12.8v-70.4c0-6.4-6.4-12.8-12.8-12.8h-38.4c-6.4 0-12.8 6.4-12.8 12.8v70.4c0 6.4 6.4 12.8 12.8 12.8zm96 0h38.4c6.4 0 12.8-6.4 12.8-12.8V108.8c0-6.4-6.4-12.8-12.8-12.8h-38.4c-6.4 0-12.8 6.4-12.8 12.8v198.4c0 6.4 6.4 12.8 12.8 12.8zM496 384H64V80c0-8.84-7.16-16-16-16H16C7.16 64 0 71.16 0 80v336c0 17.67 14.33 32 32 32h464c8.84 0 16-7.16 16-16v-32c0-8.84-7.16-16-16-16z"
                />
              </svg>
              <h3>
                Unidade consumidora
              </h3>
            </a>
          </li>
          <li
            class="sc-eDnVMP bhYUgY"
          >
            <a
              class="link"
              href="/register_month"
            >
              <svg
                class="cog"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M487.4 315.7l-42.6-24.6c4.3-23.2 4.3-47 0-70.2l42.6-24.6c4.9-2.8 7.1-8.6 5.5-14-11.1-35.6-30-67.8-54.7-94.6-3.8-4.1-10-5.1-14.8-2.3L380.8 110c-17.9-15.4-38.5-27.3-60.8-35.1V25.8c0-5.6-3.9-10.5-9.4-11.7-36.7-8.2-74.3-7.8-109.2 0-5.5 1.2-9.4 6.1-9.4 11.7V75c-22.2 7.9-42.8 19.8-60.8 35.1L88.7 85.5c-4.9-2.8-11-1.9-14.8 2.3-24.7 26.7-43.6 58.9-54.7 94.6-1.7 5.4.6 11.2 5.5 14L67.3 221c-4.3 23.2-4.3 47 0 70.2l-42.6 24.6c-4.9 2.8-7.1 8.6-5.5 14 11.1 35.6 30 67.8 54.7 94.6 3.8 4.1 10 5.1 14.8 2.3l42.6-24.6c17.9 15.4 38.5 27.3 60.8 35.1v49.2c0 5.6 3.9 10.5 9.4 11.7 36.7 8.2 74.3 7.8 109.2 0 5.5-1.2 9.4-6.1 9.4-11.7v-49.2c22.2-7.9 42.8-19.8 60.8-35.1l42.6 24.6c4.9 2.8 11 1.9 14.8-2.3 24.7-26.7 43.6-58.9 54.7-94.6 1.5-5.5-.7-11.3-5.6-14.1zM256 336c-44.1 0-80-35.9-80-80s35.9-80 80-80 80 35.9 80 80-35.9 80-80 80z"
                />
              </svg>
              <h3>
                Cadastro de en√©rgia gerada
              </h3>
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </aside>
</DocumentFragment>
`;
